src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { deleteTodos, updateTodoOnServer } from '../../api/todos';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { ChangeFunction } from '../../types/ChangeFunction';
+
+type Props = {
+  todo: Todo;
+  isLoading: boolean;
+  DeleteTodo?: (todoId: number) => void;
+  ChangeTodo?: ChangeFunction;
+  showError?: (errorType: ErrorMessage) => void;
+  hideError?: () => void;
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  isLoading,
+  DeleteTodo = () => {},
+  ChangeTodo = () => {},
+  showError = () => {},
+  hideError = () => {},
+}) => {
+  const { title, completed, id } = todo;
+
+  const [isEdited, setIsEdit] = useState(false);
+  const [isWaiting, setIsWaiting] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(title);
+
+  const editFormRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (editFormRef.current && isEdited) {
+      editFormRef.current.focus();
+    }
+  }, [isEdited]);
+
+  const handleDeleteTodo = async (onError?: () => void) => {
+    hideError();
+    setIsWaiting(true);
+
+    try {
+      await deleteTodos(id);
+      DeleteTodo(id);
+    } catch {
+      showError(ErrorMessage.Delete);
+      setIsWaiting(false);
+
+      onError?.();
```

Comment:
will be better to set some boolean variable and depending on it call / not call some function